src/app/WriteClient.cpp

```diff
@@ -482,20 +494,39 @@ CHIP_ERROR WriteClient::OnMessageReceived(Messaging::ExchangeContext * apExchang
     }
     else if (aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
     {
-        err = StatusResponse::ProcessStatusResponse(std::move(aPayload));
-        SuccessOrExit(err);
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        SuccessOrExit(err = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError));
+        suppressErrorStatusResponse = true;
+        SuccessOrExit(err = statusError);
     }
     else
     {
         err = CHIP_ERROR_INVALID_MESSAGE_TYPE;
     }
 
 exit:
-    if (mpCallback != nullptr)
+    return ResponseMessageHandled(err, apExchangeContext, suppressErrorStatusResponse);
+}
+
+CHIP_ERROR WriteClient::ResponseMessageHandled(CHIP_ERROR aError, Messaging::ExchangeContext * apExchangeContext,
```

Comment:
Again, I suspect this could just return void.  No one cares what it returns...  Either way, though; at least we are not mixing up our two errors in the body....

Reply:
updated, thanks